app/generic/verifier/verifier.py

```diff
@@ -36,6 +39,103 @@ def __get_verifier_by_version(protocol_version: str) -> Verifier:
         raise ValueError(f"Unknown protocol version {protocol_version}")
 
 
+@router.post("/credentials")
+async def get_credentials(
+    proof_request: ProofRequestGeneric,
+    aries_controller: AcaPyClient = Depends(agent_selector),
+) -> List[IndyCredPrecis]:
+    """
+     Get matching credentials for  presentation exchange
+
+     Parameters:
+     ----------
+    proof_request: ProofRequestGeneric
+         The proof request object
+
+     Returns:
+     --------
+     presentation_exchange_list: [IndyCredPrecis]
+         The list of Indy presentation credentials
+    """
+    try:
+        prover = __get_verifier_by_version(
+            protocol_version=proof_request.protocol_version
+        )
+        return await prover.get_creds(
+            controller=aries_controller, pres_ex_id=proof_request.proof_id
+        )
+    except Exception as e:
+        logger.error(
+            f"Failed to get matching credentials: \n {proof_request.proof_id}\n \n{e!r}"
+        )
+        raise e from e
+
+
+@router.post("/proofs")
+async def get_proofs(
+    proof_request: ProofRequestGeneric,
+    aries_controller: AcaPyClient = Depends(agent_selector),
+) -> List[PresentationExchange]:
+    """
+    Get a specific proof record
+
+    Parameters:
+    ----------
+    proof_request: ProofRequestGeneric
+        The proof request object
+
+    Returns:
+    --------
+    presentation_exchange_list: PresentationExchange
+        The of presentation exchange record for the proof ID
+    """
+    try:
+        prover = __get_verifier_by_version(
+            protocol_version=proof_request.protocol_version
+        )
+        # If a proof ID os provided fetch a single record for tht ID
+        if proof_request.proof_id:
+            return await prover.get_proofs(
+                controller=aries_controller, pres_ex_id=proof_request.proof_id
+            )
+        # Otherwise fecth all records for the wallet
+        else:
+            return await prover.get_proofs(controller=aries_controller)
+    except Exception as e:
+        logger.error(f"Failed to get proof(s): \n{e!r}")
+        raise e from e
+
+
+@router.delete("/proofs/{proof_id}")
+async def delete_proof(
+    proof_id: str,
+    aries_controller: AcaPyClient = Depends(agent_selector),
+) -> None:
+    """
+    Delete proofs record for proof_id (pres_ex_id including prepending version hint 'v1-' or 'v2-')
+
+    Parameters:
+    ----------
+    proof_id: str
+        The proof ID - starting with v1- or v2-
+
+    Returns:
+    --------
+    None
+    """
+    try:
+        # Currently proof_id[0:2] is required for otherwise the protocol version is unknown
+        # Maybe there is a better way to handle this?
+        # Possibly also handling proof-di not starting with v1- or v2-?
+        # Would it make sense to assume it is v1- by default/if none is supplied
+        prover = __get_verifier_by_version(protocol_version=proof_id[0:2])
```

Comment:
Maybe we can add an additional util method `__get_verifier_by_id` that does this internally?